components/datadog/agent/ecsFargate.go

```diff
@@ -48,3 +49,45 @@ func ECSFargateLinuxContainerDefinition(e config.CommonEnvironment, apiKeySSMPar
 		VolumesFrom:      ecs.TaskDefinitionVolumeFromArray{},
 	}
 }
+
+func ECSFargateLinuxContainerDefinitionLatest(apiKeySSMParamName pulumi.StringInput, fakeintake *ddfakeintake.ConnectionExporter, logConfig ecs.TaskDefinitionLogConfigurationPtrInput) *ecs.TaskDefinitionContainerDefinitionArgs {
```

Comment:
Let's just take `image` as parameter and fallback to `pulumi.Sprintf(DockerAgentFullImagePath(&e, "public.ecr.aws/datadog/agent"))` if parameter is empty.

Reply:
I added an `image` parameter to the `ECSFargateLinuxContainerDefinition`. It may be better to do it with Optional function parameter so we can omit the `image` argument when calling the function

Reply:
I am not sure we want to pass optional in internal APIs. It adds a lot of overhead for low value and we cannot pass `Pulumi.ResourceOption` anymore. I think that passing `struct` args around is the most concise and fits well with Pulumi patterns.